src/lib/router.js

```diff
@@ -33,6 +33,23 @@ async function getPage(url) {
   return domparser.parseFromString(text, "text/html");
 }
 
+/**
+ * Force the user's cursor to the target element.
+ *
+ * @param {?Element} el
+ */
+function forceFocus(el) {
+  if (!el) {
+    // do nothing
+  } else if (el.hasAttribute("tabindex")) {
+    el.focus();
+  } else {
+    el.tabIndex = -1;
+    el.focus();
+    el.deleteAttribute("tabindex");
```

Comment:
```suggestion
    el.removeAttribute("tabindex");
```

Reply:
I just tested this and removing the attribute causes focus to blur and go back to the body.

If you want to suppress the focus ring, I think instead of removing the attribute we could add a class, something like `.w-force-focus` which sets `outline: 0 !important;`

We could then either leave the class in place, or add a `once` listener to the element's blur event to remove the class.

Reply:
So you're not wrong but _keyboard focus_ now starts from this location. I've used this trick in a few places (`deleteAttribute` being wrong notwithstanding 🤣) and I always thought it was good enough.

Isn't it weird to have a focused thing with no highlight? Can I set some aria-property to imply that focus is here instead, at least until I focus on something new?

Reply:
> So you're not wrong but keyboard focus now starts from this location. I've used this trick in a few places (deleteAttribute being wrong notwithstanding 🤣) and I always thought it was good enough.

Yeah it may move the focus navigation start point, but the screen reader cursor doesn't go to the header. I think it actually goes to the nearest landmark which is main in this case. So if you turn on a screen reader and click a link, instead of reading the header text (which is nice) it says "main". IOW, you've improved the experience for folks using keyboard and switch devices, but not for folks using screen readers.

> Isn't it weird to have a focused thing with no highlight?

Managing focus like you're doing in this case is basically the one exception where you're allowed to put "focus" on something and hide the ring. The ring is meant to indicate that the user can interact with something. In this case we're putting focus on a non-interactive header to both move the focus navigation start point and to have a screen reader tell the user that new content has loaded.

> Can I set some aria-property to imply that focus is here instead, at least until I focus on something new?

I don't think so. ARIA technically doesn't affect focus behavior at all. It's just meant to add semantic info to the tree. I think you might be thinking of things like [aria-owns](https://developers.google.com/web/fundamentals/accessibility/semantics-aria/aria-labels-and-relationships#aria-owns) where you can kind of say "hey even though this element isn't my child, pretend like it is."

Reply:
I've incorporated your suggested changes now!